xbatcher/generators.py

```diff
@@ -105,6 +117,7 @@ def __init__(
         batch_dims={},
         concat_input_dims=False,
         preload_batch=True,
+        squeeze_batch_dim=True,
```

Comment:
Is it ok to set the default to `True`? I.e. squeeze the batch dimension if all dims are included in the input dims. This might break existing behaviour, so wondering if the default should be `False` instead for backward compatibility.

Reply:
It would definitely break existing behavior. I'm not really sure which would be more intuitive, I think that's kind of subjective, but personally I would agree with you.

Reply:
> The problem is that the existing behavior _already_ squeezes the array, so I think it would be kind of a pain to have to use `xr.DataArray.expand_dims`,

Yeah, I also find this unintuitive, the fact that xbatcher squeezes any non `input_dims` into a dim called `batch`. So another way to think of this `squeeze_batch_dim` feature is that xbatcher will just return the cropped/sliced/chipped array without squeezing the dims. This is more important with higher dimensional (＞3D) arrays because sometimes you do want to reserve the extra dims.